internal/observability/event/hclog_event_adapter.go

```diff
@@ -0,0 +1,235 @@
+package event
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"sync"
+
+	"github.com/hashicorp/go-hclog"
+)
+
+// HclogLoggerAdapter is used to provide an hclog-style interface to code that
+// cannot natively handle eventing. Currently, all log lines are written as
+// system events. Note that this is not meant for high throughput; some
+// potential optimizations (such as using atomic values for name and such) are
+// not current implemented. Additionally, some functions (such as fetching a
+// stdlib logger/writer) are simply not supported right now.
+type HclogLoggerAdapter struct {
+	eventCtx context.Context
+	l        *sync.RWMutex
+	level    hclog.Level
+	name     string
+	withArgs []interface{}
+}
+
+// Ensure that we are implementing Logger
+var _ hclog.Logger = (*HclogLoggerAdapter)(nil)
+
+func HclogLogger(ctx context.Context, e *Eventer, opt ...Option) (hclog.Logger, error) {
+	const op = "event.HclogLogger"
+	eventCtx, err := NewEventerContext(ctx, e)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", op, err)
+	}
+	opts := getOpts(opt...)
+	return &HclogLoggerAdapter{
+		eventCtx: eventCtx,
+		l:        new(sync.RWMutex),
+		level:    opts.withHclogLevel,
+	}, nil
+}
+
+// Args are alternating key, val pairs
+// keys must be strings
+// vals can be any type, but display is implementation specific
+// Emit a message and key/value pairs at a provided log level
+func (h *HclogLoggerAdapter) Log(level hclog.Level, msg string, args ...interface{}) {
+	switch {
+	case h.level == hclog.NoLevel: // If logger is not set to any level, accept it
+	case h.level <= level: // Otherwise if logger is same or more verbose, accept
+	default:
+		return
+	}
+	h.writeEvent("", msg, args)
+}
+
+// Emit a message and key/value pairs at the TRACE level
+func (h *HclogLoggerAdapter) Trace(msg string, args ...interface{}) {
+	if h.level > hclog.Trace {
+		return
+	}
+	h.writeEvent("", msg, args)
+}
+
+// Emit a message and key/value pairs at the DEBUG level
+func (h *HclogLoggerAdapter) Debug(msg string, args ...interface{}) {
+	if h.level > hclog.Debug {
+		return
+	}
+	h.writeEvent("", msg, args)
+}
+
+// Emit a message and key/value pairs at the INFO level
+func (h *HclogLoggerAdapter) Info(msg string, args ...interface{}) {
```

Comment:
Most of the comment in this file do not start with the name of the function as godoc suggests

Reply:
Yeah, they are copied verbatim from the `hclog.Logger` interface. Maybe they shouldn't be? 🤷 But going to leave it that way for the moment.